operators/pkg/controller/apmserver/config/config.go

```diff
@@ -101,5 +101,9 @@ func NewConfigFromSpec(c k8s.Client, as v1alpha1.ApmServer) (*settings.Canonical
 		),
 		userSettings,
 	)
+
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Nitpick: Most of the time we prefer to stick the "check and return error block" to the previous block without a separated blank line. Quite often we use a blank line above the `return` but I think that's up to you. I think that can increase readability a little, especially when the functions reach more than 30 lines.